pkg/netpol/connlist/connlist_test.go

```diff
@@ -1,44 +1,96 @@
-		analyzer = NewConnlistAnalyzer(WithStopOnError())
-	} else {
-		analyzer = NewConnlistAnalyzer()
-	}
+const connlistExpectedOutputFileNamePrefix = "connlist_output."
+const underscore = "_"
+
+var allFormats = []string{common.TextFormat, common.JSONFormat, common.CSVFormat, common.MDFormat, common.DOTFormat}
 
-	res, _, err := analyzer.ConnlistFromDirPath(path)
+// helping func - returns test's dir path from test's dir name
+func getDirPathFromDirName(dirName string) string {
+	return filepath.Join(testutils.GetTestsDir(), dirName)
+}
+
+// helping func - creates ConnlistAnalyzer with desired opts and returns the analyzer with connlist from provided directory
+func getConnlistFromDirPathRes(opts []ConnlistAnalyzerOption, dirName string) (*ConnlistAnalyzer, []Peer2PeerConnection, error) {
+	analyzer := NewConnlistAnalyzer(opts...)
+	res, _, err := analyzer.ConnlistFromDirPath(getDirPathFromDirName(dirName))
 	return analyzer, res, err
 }
 
-type testEntry struct {
-	testDirName          string
-	outputFormats        []string
-	generateActualOutput bool // if true, overrides existing expected output file
+// helping func - creates the analyzer , gets connlist and writes it to string and verifies results
+func verifyConnlistAnalyzeOutputVsExpectedOutput(t *testing.T, analyzerOptions []ConnlistAnalyzerOption, dirName,
+	expectedOutputFileName, testName, format string) {
+	analyzer, res, err := getConnlistFromDirPathRes(analyzerOptions, dirName)
+	require.Nil(t, err, getDebugMsgWithTestNameAndFormat(testName, format))
+	output, err := analyzer.ConnectionsListToString(res)
+	require.Nil(t, err, getDebugMsgWithTestNameAndFormat(testName, format))
+	checkActualVsExpectedOutputMatch(t, testName, dirName, expectedOutputFileName, output, format)
+}
+
+// helping func - writes debug message for good path tests
+func getDebugMsgWithTestNameAndFormat(testName, format string) string {
+	return fmt.Sprintf("test: %q, output format: %q", testName, format)
 }
 
-const expectedOutputFileNamePrefix = "connlist_output."
+// helping func - checks if actual output matches expected output, if not generates actual output file
+func checkActualVsExpectedOutputMatch(t *testing.T, testName, dirName, expectedOutputFileName, actualOutput, format string) {
+	actualOutputFileName := "actual_" + expectedOutputFileName
+	// read expected output file
+	expectedOutputFile := filepath.Join(getDirPathFromDirName(dirName), expectedOutputFileName)
+	expectedOutput, err := os.ReadFile(expectedOutputFile)
+	require.Nil(t, err, getDebugMsgWithTestNameAndFormat(testName, format))
+	actualOutputFile := filepath.Join(getDirPathFromDirName(dirName), actualOutputFileName)
+	if string(expectedOutput) != actualOutput {
+		// generate actual output file
+		fp, err := os.Create(actualOutputFile)
+		require.Nil(t, err, getDebugMsgWithTestNameAndFormat(testName, format))
+		_, err = fp.WriteString(actualOutput)
+		require.Nil(t, err, getDebugMsgWithTestNameAndFormat(testName, format))
```

Comment:
There is already a function `writeRes` in `eval_test` for writing to a file. Consider moving it to a common pkg and use the same function here as well, rather than having multiple variations for this code.

Reply:
done some changes.
more updates in `eval_test.go` will be done on the eval test improvements branch

Reply:
Note that I've updated the issue description - with list of packages to handle and priorities.
So I suggest waiting with `eval` pkg tests.